staging/src/k8s.io/client-go/tools/clientcmd/loader_test.go

```diff
@@ -146,6 +148,39 @@ func TestErrorReadingNonFile(t *testing.T) {
 	}
 }
 
+func TestErrorWrapping(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Couldn't create tmpdir")
+	}
+	defer os.RemoveAll(tmpdir)
+
+	loadingRules := ClientConfigLoadingRules{
+		ExplicitPath: tmpdir,
+	}
+
+	_, err = loadingRules.Load()
+	if err == nil {
+		t.Fatal("Expected error for non-file, got none")
+	}
+
+	aggErrors, ok := err.(utilerrors.Aggregate)
+	if !ok {
+		t.Fatal("Incorrect error type returned, was not an aggregate")
+	}
+
+	onlyError := utilerrors.Reduce(aggErrors)
+
+	finalError := errors.Unwrap(onlyError)
+	if err == nil {
+		t.Fatal("Expected unwrap to succeed")
+	}
+
+	if _, ok := finalError.(*os.PathError); !ok {
+		t.Fatal("Error type could not be validated")
+	}
+}
```

Comment:
Also try the `errors.Is(onlyError, os.ErrNotExist)`?

Reply:
Also ther `os.IsNotExist(finalError)`?

Reply:
Hey @caesarxuchao - don't really understand what you are asking for test wise here? This isn't meant to be an exhaustive test of all errors and whether they unwrap just whether or not a concrete error type can be unwrapped. Since `os.PathError` succeeds I'd expected `os.ErrNotExist` or any other concrete error type to pass as well (without a specific test for each).

Am I misunderstanding what you are asking for here?

Reply:
I'm not trying to test the concrete error types. I want to try these different error assertion methods. `errors.Is` is the new golang way to test a wrapped error. os.IsNotExist is provided by the library. Does that make sense?

Reply:
I think I got it now, You are just saying instead of an `Unwrap` and type assertion, just use `errors.Is`.

Reply:
alright - I dug into this a bit and the `Unwrap` and type assertion is as good as this test can get. The error that actually comes back is `PathError` due to the fact that we attempt to open a Directory as a File (which gives us `read /var/blah is a directory` this directly maps to a `PathError` which is a struct type that you can pass to `errors.Is` but it will not much the returned error.